apps/ejabberd/src/ejabberd_c2s.erl

```diff
@@ -3087,6 +3146,7 @@ defer_resource_constraint_check(#state{stream_mgmt_constraint_check_tref = undef
 defer_resource_constraint_check(State)->
     State.
 
+-spec sasl_success_stanza(any()) -> xmlel().
```

Comment:
I personally think the type change should go in it's own commit. There is no need to mix this with other changes (unless the changes are needed to fix dialyzer errors).